src/gluonnlp/utils/misc.py

```diff
@@ -651,3 +651,24 @@ def init_comm(backend, gpus):
             logging.info('GPU communication supported by KVStore')
 
     return store, num_workers, rank, local_rank, is_master_node, ctx_l
+
+
+def get_mxnet_recommended_ctx():
```

Comment:
When committing to this function as part of 1.0 API, should we choose a more descriptive name? I think "recommended" is a bit unclear, as the function doesn't really know which context will work best for the particular setting. Also you're returning a list of ctx, not ctx.

Reply:
Do you have any suggestions about the name?

Reply:
Or `get_mxnet_all_ctx()`?

Reply:
`get_mxnet_gpu_ctx_list_fallback_cpu`?

Reply:
This looks too long.

Reply:
All we still use get_mxnet_visible_ctx ?